quint/src/parsing/ToIrListener.ts

```diff
@@ -220,89 +219,79 @@ export class ToIrListener implements QuintListener {
   /** **************** translate operator definititons **********************/
 
   // translate a top-level or nested operator definition
-  exitOperDef(ctx: p.OperDefContext) {
+  exitAnnotatedOperDef(ctx: p.AnnotatedOperDefContext) {
     const name = ctx.normalCallName().text
-    const [params, typeTag] = this.processOpDefParams(ctx)
-    // get the definition body
+
+    const params = ctx._annotOperParam
+      .map(_ => popOrFail(this.paramStack, 'violated grammar of annotated AnnotatedOperDef'))
+      .reverse()
+    const res = this.popType().unwrap(() => 'violated grammar of annotated params return type')
+    const args = params.map(p => {
+      assert(isAnnotatedDef(p), 'violated grammar of annotated param type')
```

Comment:
These unwraps and asserts are just like the ones in all the `exitType*` methods: the grammar will not bring us to visit this context without these pre-requisites unless we've messed up something badly. It should have no impact on incremental parsing, unless we have a bigger problem that we want to know about.